pyomo/gdp/plugins/fix_disjuncts.py

```diff
@@ -38,20 +43,33 @@ def __init__(self, **kwargs):
     ))
 
     def _apply_to(self, model, **kwds):
-        """Fix all disjuncts in the given model and reclassify them to Blocks."""
+        """Fix all disjuncts in the given model and reclassify them to 
+        Blocks."""
         config = self.config = self.CONFIG(kwds.pop('options', {}))
         config.set_value(kwds)
 
         self._transformContainer(model)
 
         # Reclassify all disjuncts
-        for disjunct_object in model.component_objects(Disjunct, descend_into=(Block, Disjunct)):
-            disjunct_object.parent_block().reclassify_component_type(disjunct_object, Block)
+        for disjunct_object in model.component_objects(Disjunct,
+                                                       descend_into=(Block,
+                                                                     Disjunct)):
+            disjunct_object.parent_block().reclassify_component_type(
+                disjunct_object, Block)
+
+        # Transform any remaining logical stuff
+        TransformationFactory('core.logical_to_linear').apply_to(model)
 
     def _transformContainer(self, obj):
         """Find all disjuncts in the container and transform them."""
-        for disjunct in obj.component_data_objects(ctype=Disjunct, active=True, descend_into=True):
+        for disjunct in obj.component_data_objects(ctype=Disjunct, active=True,
+                                                   descend_into=True):
             _binary = disjunct.binary_indicator_var
```

Comment:
This transformation predates the conversion of `indicator_var` to being a `BooleanVar`.  I believe that this logic can be simplified to use `indicator_var` and only check for `None`, `True`, and `False`.  Fractional `binary_indicator_var` values are automatically mapped to `None` values for `indicator_var`.  I would recommend switching to `indicator_var` and removing the tolerance logic below.

Reply:
OK, I changed this. I removed the integer_tolerance option completely because there's no way to change that tolerance for the BooleanVars right now, I don't think.